hips/tiles/tile.py

```diff
@@ -0,0 +1,131 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from astropy.utils.data import get_pkg_data_filename
+from .description import HipsDescription
+from astropy.io import fits
+from pathlib import Path
+from io import BytesIO
+from PIL import Image
+import numpy as np
+import urllib.request
+
+__all__ = [
+    'HipsTile',
+]
+
+
+class HipsTile:
+    """HiPS tile container.
+
+    This class provides methods for fetching, reading,
+    and stroing a HiPS tile. It also contains a few
+    getters and setters around most commonly used
+    HiPS tile attributes.
+
+    Parameters
+    ----------
+    hips_description : `HipsDescription`
+        Class HipsDescription contains HiPS properties
+    order : `int`
+        Order of the HiPS
+    ipix : `int`
+        HEALPix pixel number
+    tile_width : `int`
+        Width of HiPS tile
+    format : `format`
+        Format of HiPS tile
+    data : `list`
+        Pixel values of HiPS tile
+
+    Examples
+    --------
+    ::
+
+        >>> from hips.tiles import HipsDescription
+        >>> from astropy.utils.data import get_pkg_data_filename
+        >>> filename = get_pkg_data_filename('tests/data/properties.txt')
+        >>> hips_description = HipsDescription.read(filename)
+        >>> hips_tile = HipsTile(hips_description, 6, 24185, 'fits')
+        >>> tile.read()
+        >>> tile.data
+        [[3617 3304 4196 ..., 4545 4800 5701]
+         [3638 3288 3934 ..., 4448 5029 4951]
+         [3116 3653 3412 ..., 5282 4406 4967]
+         ...,
+         [3890 3463 3448 ..., 5980 6902 6543]
+         [4845 4065 3821 ..., 6928 6557 7197]
+         [4261 3581 3842 ..., 7095 6390 6191]]
+    """
+
+    def __init__(self, hips_description: HipsDescription, order: int, ipix: int, format: str, data: list=None, tile_width: int=512) -> None:
+        self.hips_description = hips_description
+        self.order = order
+        self.ipix = ipix
+        self.tile_width = tile_width
+        self.format = format
+        self.data = data
+
+    @staticmethod
+    def _directory(ipix: int) -> int:
+        """Directory of the HiPS tile (`int`)."""
+        return np.around(ipix, decimals=-(len(str(ipix)) - 1))
+
+    @property
+    def base_url(self) -> str:
+        """HiPS tile base url (`str`)."""
+        return self._base_url
```

Comment:
I was assigning the `base_url` after initialization through `@base_url.setter`, and accessing it through `@property`. But, I have removed the setter and getter now as this can be done without that as well.